src/libs/antares/study/parts/common/cluster_list.h

```diff
@@ -190,76 +133,71 @@ class ClusterList
     */
     bool forceReload(bool reload = false) const;
 
     /*!
-    ** \brief Mark the cluster as modified
+    ** \brief Mark the clusters as modified
     */
     void markAsModified() const;
 
-    /*!
-    ** \brief Rebuild the index of clusters
-    **
-    ** As a list of clusters is a hash table, it is not
-    ** possible to directly accees to a cluster from its index.
-    ** However an index can be built but it must be re-built when
-    ** the hash table is modified.
-    */
-    void rebuildIndex();
 
     /*!
     ** \brief Get the size (bytes) occupied in memory by a `ClusterList` structure
     ** \return A size (in bytes)
     */
     uint64_t memoryUsage() const;
 
-public:
-    //! All clusters by their index
-    std::vector<ClusterT*> byIndex;
-    //! All clusters
-    Map cluster;
-
-    // thermal, renewable, etc.
-    virtual YString typeID() const = 0;
-
-    /*!
-    ** \brief Number of dispatchable cluster per group
-    **
-    ** You should rely on these values only after the loading of the study
-    ** and until the study is not modified.
-    ** These values are modified by 'ClusterListAdd()'
-    */
-    std::vector<uint> groupCount;
-
-    int loadDataSeriesFromFolder(Study& study,
+    /// \name IO functions
+    /// @{
+    bool loadDataSeriesFromFolder(Study& study,
                                  const StudyLoadOptions& options,
                                  const AnyString& folder);
 
-    int saveDataSeriesToFolder(const AnyString& folder) const;
+    bool saveDataSeriesToFolder(const AnyString& folder) const;
 
-    int saveDataSeriesToFolder(const AnyString& folder, const YString& msg) const;
+    bool saveDataSeriesToFolder(const AnyString& folder, const YString& msg) const;
 
     virtual bool saveToFolder(const AnyString& folder) const = 0;
+    ///@}
 
-    //! \name Informations
-    //@{
     /*!
     ** \brief Retrieve the total capacity and the total unit count
     **
     ** Pseudo code:
     ** \code
-    ** each thermal cluster do
-    ** 	total += cluster{unit count} * cluster{nominal capacity}
-    **	unit  += cluster{unit count}
+    ** each thermal clusters do
+    ** 	total += clusters{unit count} * clusters{nominal capacity}
+    **	unit  += clusters{unit count}
     ** \endcode
     */
     void retrieveTotalCapacityAndUnitCount(double& total, uint& unitCount) const;
-    //@}
+
+    /*!
+    ** \brief Removes disabled clusters
+    **
+    ** All clusters with the flag 'enabled' turned to false will be removed from 'list'.
+    ** As a consequence, they will no longer be seen as thermal clusters
+    ** from the solver's point of view.
+    ** \warning This method should only be used from the solver
+    **
+    ** \return The number of disabled clusters found
+    */
+    uint removeDisabledClusters();
+
+protected:
+    /// The vector containing the clusters
+    Vect clusters;
+
+private:
+    /// thermal, renewable, etc.
+    std::string typeID = "common";
```

Comment:
This won't work. We need dynamic polymorphism for this, and static member variables can't have that. Please revert to `virtual` member function.